include/zephyr/drivers/gpio.h

```diff
@@ -45,24 +45,24 @@ extern "C" {
  */
 
 /** Enables pin as input. */
-#define GPIO_INPUT              (1U << 16)
+#define GPIO_INPUT              (1U << 16U)
```

Comment:
I'm not sure this is correct. godbolt says that unsigned shifted by int is still unsigned:

https://godbolt.org/z/MG9on18Ws

And this reference:

https://learn.microsoft.com/en-us/cpp/cpp/left-shift-and-right-shift-operators-input-and-output?view=msvc-170#shifts-and-promotions

says that the resulting type is the result of promoting (unsigned, int) to a common type, which is unsigned.

I also can't find any other use of this in Zephyr.

So I don't think this removes undefined behaviour, but it does make it mildly harder to read.

Reply:
Well, both links doesn't really help. It just shows how a specific compiler behaves, therefore "implementation defined".

Reply:
Sure, for the first, but the second is Microsoft's summary of the standard, not their implementation.

Here's the relevant section from cppreference:

https://en.cppreference.com/w/cpp/language/operator_arithmetic "Built-in bitwise shift operators": "Integral promotions are performed on both operands." and "The type of the result is that of lhs."

I couldn't find a copy of the C spec to reference.